packages/flutter_hooks/lib/src/framework.dart

```diff
@@ -354,6 +373,7 @@ mixin HookElement on ComponentElement {
   _Entry<HookState>? _currentHookState;
   final _hooks = LinkedList<_Entry<HookState>>();
   final _shouldRebuildQueue = LinkedList<_Entry<bool Function()>>();
+  Set<HookState>? _needDidBuildHooks;
```

Comment:
Should be a list, not a set

The setter should be able to implement all the logic to avoid duplicate states.

Reply:
why doing that manually when the `Set` does it for us ? Because doing the manually on the setter is more code to test and maintain. What is the gain here ?

Reply:
Performance

List has a more efficient add & iteration

Reply:
Considering the deduplication logic doesn't require a `contains` call because the hook state knows if it was added to the element or not, using a List instead of a Set is a significant boost.

Reply:
Changed to List then :) thanks for the explanation!